pkg/apis/apm/v1beta1/apmserver_types.go

```diff
@@ -0,0 +1,139 @@
+	Version string `json:"version,omitempty"`
+
+	// Image represents the docker image that will be used.
+	Image string `json:"image,omitempty"`
+
+	// NodeCount defines how many nodes the Apm Server deployment must have.
+	NodeCount int32 `json:"nodeCount,omitempty"`
+
+	// Config represents the APM configuration.
+	Config *commonv1beta1.Config `json:"config,omitempty"`
+
+	// HTTP contains settings for HTTP.
+	HTTP commonv1beta1.HTTPConfig `json:"http,omitempty"`
+
+	// ElasticsearchRef references an Elasticsearch resource in the Kubernetes cluster.
+	// If the namespace is not specified, the current resource namespace will be used.
+	ElasticsearchRef commonv1beta1.ObjectSelector `json:"elasticsearchRef,omitempty"`
+
+	// PodTemplate can be used to propagate configuration to APM Server pods.
+	// This allows specifying custom annotations, labels, environment variables,
+	// affinity, resources, etc. for the pods created from this NodeSpec.
+	// +kubebuilder:validation:Optional
+	PodTemplate corev1.PodTemplateSpec `json:"podTemplate,omitempty"`
+
+	// SecureSettings references secrets containing secure settings, to be injected
+	// into the APM keystore on each node.
+	// Each individual key/value entry in the referenced secrets is considered as an
+	// individual secure setting to be injected.
+	// You can use the `entries` and `key` fields to consider only a subset of the secret
+	// entries and the `path` field to change the target path of a secret entry key.
+	// The secret must exist in the same namespace as the APM resource.
+	SecureSettings []commonv1beta1.SecretSource `json:"secureSettings,omitempty"`
+}
+
+// ApmServerHealth expresses the status of the Apm Server instances.
+type ApmServerHealth string
+
+const (
+	// ApmServerRed means no instance is currently available.
+	ApmServerRed ApmServerHealth = "red"
+	// ApmServerGreen means at least one instance is available.
+	ApmServerGreen ApmServerHealth = "green"
+)
+
+// ApmServerStatus defines the observed state of ApmServer
+type ApmServerStatus struct {
+	commonv1beta1.ReconcilerStatus `json:",inline"`
+	Health                         ApmServerHealth `json:"health,omitempty"`
+	// ExternalService is the name of the service the agents should connect to.
+	ExternalService string `json:"service,omitempty"`
+	// SecretTokenSecretName is the name of the Secret that contains the secret token
+	SecretTokenSecretName string `json:"secretTokenSecret,omitempty"`
+	// Association is the status of any auto-linking to Elasticsearch clusters.
+	Association commonv1beta1.AssociationStatus `json:"associationStatus,omitempty"`
+}
+
+// IsDegraded returns true if the current status is worse than the previous.
+func (as ApmServerStatus) IsDegraded(prev ApmServerStatus) bool {
+	return prev.Health == ApmServerGreen && as.Health != ApmServerGreen
+}
+
+// +kubebuilder:object:root=true
+
+// ApmServer is the Schema for the apmservers API
+// +kubebuilder:categories=elastic
```

Comment:
We used to be able to do `kubectl get elastic` but it looks like it does not work anymore with the new CRDs.

Reply:
I'm investigating.

Reply:
Looks like the new marker is groupName
https://book.kubebuilder.io/reference/markers/crd.html

Reply:
hm.. or is it `// +kubebuilder:resource:categories=elastic` (instead of `// +kubebuilder:categories=elastic`)?

Reply:
Not necessary to this PR, but I wonder if we also want to add it to the `all` category

Reply:
Well seen. I observed that but forgot to have a look.

Reply:
Ok. The annotation is now `kubebuilder:resource:categories`.